File: Composicion y Agregacion/ejercicio 5/Python/Jugador.py
class Jugador:
    def __init__(self, nombre, numero, posicion):
        self.nombre = nombre
        self.numero = numero
        self.posicion = posicion
    def mostrar(self):
        print("Jugador:", self.nombre)
        print("Número:", self.numero)
        print("Posición:", self.posicion)
class Portero(Jugador):
    def __init__(self, nombre, numero, habilidad):
        super().__init__(nombre, numero, "Portero")
        self.habilidad = habilidad
    def mostrar(self):
        super().mostrar()
        print("Habilidad especial:", self.habilidad)
class Defensa(Jugador):
    def __init__(self, nombre, numero, habilidad):
        super().__init__(nombre, numero, "Defensa")
        self.habilidad = habilidad
    def mostrar(self):
        super().mostrar()
        print("Habilidad especial:", self.habilidad)
class Mediocampista(Jugador):
    def __init__(self, nombre, numero, habilidad):
        super().__init__(nombre, numero, "Mediocampista")
        self.habilidad = habilidad
    def mostrar(self):
        super().mostrar()
        print("Habilidad especial:", self.habilidad)
class Delantero(Jugador):
    def __init__(self, nombre, numero, habilidad):
        super().__init__(nombre, numero, "Delantero")
        self.habilidad = habilidad
    def mostrar(self):
        super().mostrar()
        print("Habilidad especial:", self.habilidad)
class Equipo:
    def __init__(self, nombre):
        self.nombre = nombre
        self.jugadores = []
    def agrega(self, jugador):
        self.jugadores.append(jugador)
    def mostrar(self):
        print("Equipo:", self.nombre)
        for j in self.jugadores:
            j.mostrar()
e = Equipo("Los Tigres")
e.agrega(Portero("Pedro", 1, "Atajadas"))
e.agrega(Defensa("Carlos", 4, "Marcaje"))
e.mostrar()

File: Composicion y Agregacion/ejercicio 5/Java/Jugador.java
import java.util.ArrayList;
class Júgador {
    public String nombre;
    public int numero;
    public String posicion;
    public Júgador(String nombre, int numero, String posicion) {
        this.nombre = nombre;
        this.numero = numero;
        this.posicion = posicion;
    }
    public void mostrar() {
        System.out.println("Júgador: " + nombre + ", Número: " + numero + ", Posición: " + posicion);
    }
}
class Portero extends Júgador {
    public String habilidad;
    public Portero(String nombre, int numero) {
        super(nombre, numero, "Portero");
        this.habilidad = "Atajadas";
    }
    @Override
    public void mostrar() {
        super.mostrar();
        System.out.println("Habilidad especial: " + habilidad);
    }
}
class Defensa extends Júgador {
    public String habilidad;
    public Defensa(String nombre, int numero) {
        super(nombre, numero, "Defensa");
        this.habilidad = "Marcaje";
    }
    @Override
    public void mostrar() {
        super.mostrar();
        System.out.println("Habilidad especial: " + habilidad);
    }
}
class Mediocampista extends Júgador {
    public String habilidad;
    public Mediocampista(String nombre, int numero) {
        super(nombre, numero, "Mediocampista");
        this.habilidad = "Pases";
    }
    @Override
    public void mostrar() {
        super.mostrar();
        System.out.println("Habilidad especial: " + habilidad);
    }
}
class Delantero extends Júgador {
    public String habilidad;
    public Delantero(String nombre, int numero) {
        super(nombre, numero, "Delantero");
        this.habilidad = "Goles";
    }
    @Override
    public void mostrar() {
        super.mostrar();
        System.out.println("Habilidad especial: " + habilidad);
    }
}
class Equipo {
    public String nombre;
    public ArrayList<Júgador> jugadores;
    public Equipo(String nombre) {
        this.nombre = nombre;
        this.jugadores = new ArrayList<>();
    }
    public void agregar(Júgador jugador) {
        jugadores.add(jugador);
    }
    public void mostrar() {
        System.out.println("Equipo: " + nombre);
        for (Júgador j : jugadores) {
            j.mostrar();
            System.out.println();
        }
    }
}
public class Jugador {
    public static void main(String[] args) {
        Equipo equipo = new Equipo("Los Invencibles");
        equipo.agregar(new Portero("Juan", 1));
        equipo.agregar(new Defensa("Carlos", 2));
        equipo.agregar(new Mediocampista("Luis", 6));
        equipo.agregar(new Delantero("Pedro", 9));
        equipo.mostrar();
    }
}
